### PR TITLE
Upgrade to spring boot rc2

### PR DESCRIPTION
**Pull request by moarychan**
## Purpose
Upgrade to Spring Cloud Azure 6.0.0-beta.4.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Run terraform script to run samples.

## What to Check
N/A

## Other Information
N/A

### CODE REVIEW

**Review by chenrujun — DISMISSED** [review]

File: aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java [hunk 1]
```diff
@@ -30,8 +29,7 @@ SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .anyRequest().permitAll()
                 .and()
             .csrf()
-                .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
-                .and()
+                .disable()
```
Comment (saragluna) [inline aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java:32]: Do we have an issue to link here?

Comment (saragluna) [reply aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java:32]: To explain why we are disabling it.

**Review by saragluna — APPROVED** [review]

File: aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java (hunk 1 above)
Comment (moarychan) [reply aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java:32]: Opened https://github.com/Azure/azure-sdk-for-java/issues/32484

Comment (saragluna) [reply aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java:32]: @moarychan, I don't see why we decided to disable this in the issue you created. Should we add some comments in your code here or add some explanation on the issue you created?

Comment (moarychan) [reply aad/spring-cloud-azure-starter-active-directory/aad-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/security/WebSecurityConfig.java:32]: Sorry, updated it.

**Review by hui1110 — APPROVED** [review]

**Pull request merged**